Marlin/src/feature/bedlevel/mbl/mesh_bed_leveling.cpp

```diff
@@ -42,8 +42,8 @@
   mesh_bed_leveling::mesh_bed_leveling() {
     for (uint8_t i = 0; i < GRID_MAX_POINTS_X; ++i)
       index_to_xpos[i] = MESH_MIN_X + i * (MESH_X_DIST);
-    for (uint8_t i = 0; i < GRID_MAX_POINTS_Y; ++i)
-      index_to_ypos[i] = MESH_MIN_Y + i * (MESH_Y_DIST);
+    for (uint8_t j = 0; j < GRID_MAX_POINTS_Y; ++j)
+      index_to_ypos[j] = MESH_MIN_Y + j * (MESH_Y_DIST);
```

Comment:
This change doesn't belong in this PR. I'd argue that it isn't even correct, since these are not nested loops.

```suggestion
    for (uint8_t i = 0; i < GRID_MAX_POINTS_Y; ++i)
      index_to_ypos[i] = MESH_MIN_Y + i * (MESH_Y_DIST);
```

Reply:
ok that makes sense, but having that most of the other for loops specifically use "i" for "x" and "j" for "y", and to argue that it *is* correct, and consistent with that logic, take a look at this for example:
in file **Marlin\src\lcd\e3v2\marlinui\ui_common.cpp**
```y
      for (uint8_t j = 0; j < (GRID_MAX_POINTS_Y); j++, y += pixels_per_y_mesh_pnt)
        for (uint8_t i = 0, x = sx; i < (GRID_MAX_POINTS_X); i++, x += pixels_per_x_mesh_pnt)
```
albeit this ex. is nested, however it uses "j" first for "y". so it makes sense to remain relative otherwise it can be very easy to be confused as to which is which.

what about in **Marlin\src\feature\encoder_i2c.cpp** 
`for (uint8_t j = 31; j > 0; j--)`

it is not nested and the others in the same file us `i`. that must be a typo then.

and in **M48.cpp** line 244, this also uses `j` for some reason.

---
<br>

off topic, but i'm unable to find a sufficient answer to this line of code in **powerless.cpp**, maybe you can help?
```y
PROCESS_SUBCOMMANDS_NOW(TS(F("M420S"), '0' + (char)info.flag.leveling, 'Z', p_float_t(info.fade, 1)));
```
this part in particular:
`TS(F("M420S"), '0' + (char)info.flag.leveling, 'Z',` 
wouldn't this result in it being "M420S0#Z...." as in (#) either '1' or '0' - the bool flag (to restore leveling)

instead would it be better to make it
`(char)(0 + info.flag.leveling), F("Z"),`

I also included the flashstringhelper for 'Z' or is this only necessary in front of actual Gcode commands?
that is another part of my question.
as you can see in this line
`PROCESS_SUBCOMMANDS_NOW(TS('T', e, 'S'))`

so like, if i were to have it like this i get a warning:
`gcode.process_subcommands_now(TS('G28XYO'));`
" warning: character constant too long for its type"